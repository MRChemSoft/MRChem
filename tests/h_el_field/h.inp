# vim:syntax=sh:

world_prec = 1.0e-3               # Overall relative precision
world_size = 5                    # Size of simulation box 2^n

MPI {
  numerically_exact = true        # Guarantee identical results in MPI
}


Molecule {
  multiplicity = 2
$coords
H       0.0000  0.0000  1.0000
$end
}

WaveFunction {
  method = HF                     # Wave function method (HF or DFT)
  restricted = false
}

ExternalFields {
  electric_field = [ 0.0, 0.0, 0.01 ]
}

Properties {
  dipole_moment = true            # Compute electrostatic dipole moment
}

SCF {
  kain = 3                        # Length of KAIN iterative history
  max_iter = 5
  guess_type = CORE_DZ
}

