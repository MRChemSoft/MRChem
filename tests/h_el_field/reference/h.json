{
  "input": {
    "molecule": {
      "charge": 0,
      "coords": [
        {
          "atom": "h",
          "xyz": [
            0.0,
            0.0,
            1.0
          ]
        }
      ],
      "multiplicity": 2
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": true,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 5,
      "basis_type": "interpolating",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -4
    },
    "printer": {
      "file_name": "h.inp",
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.001
        },
        "external_operator": {
          "electric_field": [
            0.0,
            0.0,
            0.01
          ],
          "r_O": [
            0.0,
            0.0,
            0.0
          ]
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "proj_prec": 0.001,
          "shared_memory": false,
          "smooth_prec": 0.001
        }
      },
      "initial_guess": {
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "Hartree-Fock",
        "prec": 0.001,
        "restricted": false,
        "type": "core",
        "zeta": 2
      },
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        }
      },
      "scf_solver": {
        "checkpoint": false,
        "energy_thrs": -1.0,
        "file_chk": "checkpoint/phi_scf",
        "final_prec": 0.001,
        "helmholtz_prec": -1.0,
        "kain": 3,
        "localize": false,
        "max_iter": 5,
        "method": "Hartree-Fock",
        "orbital_thrs": 0.01,
        "rotation": 0,
        "start_prec": 0.001
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        0.0,
        0.0,
        1.0
      ],
      "charge": 0,
      "dipole_moment": {
        "dip-1": {
          "magnitude": 0.03434179158539352,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            0.0,
            0.0,
            0.03434179158539352
          ],
          "vector_el": [
            0.0,
            0.0,
            -0.965658208414475
          ],
          "vector_nuc": [
            0.0,
            0.0,
            0.9999999999998685
          ]
        }
      },
      "geometry": [
        {
          "symbol": "H",
          "xyz": [
            0.0,
            0.0,
            1.0
          ]
        }
      ],
      "multiplicity": 2,
      "orbital_energies": {
        "energy": [
          -0.4912235509357724
        ],
        "occupation": [
          1.0
        ],
        "spin": [
          "a"
        ],
        "sum_occupied": -0.4912235509357724
      },
      "scf_energy": {
        "E_ee": 0.3120042667385621,
        "E_eext": 0.00965658208414476,
        "E_el": -0.49069175814205057,
        "E_en": -1.0008467878514151,
        "E_kin": 0.5010302404189421,
        "E_next": -0.009999999999998685,
        "E_nn": 0.0,
        "E_nuc": -0.009999999999998685,
        "E_tot": -0.5006917581420492,
        "E_x": -0.31253605953228436,
        "E_xc": 0.0
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 1,
      "routine": "mrchem.x",
      "total_cores": 1,
      "version": "1.1.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 0.3114941166208951,
        "E_eext": 0.010000015936173787,
        "E_el": -0.4904569092049828,
        "E_en": -1.0000031136308,
        "E_kin": 0.5000385924124842,
        "E_next": -0.009999999999998685,
        "E_nn": 0.0,
        "E_nuc": -0.009999999999998685,
        "E_tot": -0.5004569092049815,
        "E_x": -0.3119865205437357,
        "E_xc": 0.0
      },
      "scf_solver": {
        "converged": true,
        "cycles": [
          {
            "energy_terms": {
              "E_ee": 0.31190114473803987,
              "E_eext": 0.009765306929799732,
              "E_el": -0.49065400623653965,
              "E_en": -1.000659614601058,
              "E_kin": 0.5007676926341574,
              "E_next": -0.009999999999998685,
              "E_nn": 0.0,
              "E_nuc": -0.009999999999998685,
              "E_tot": -0.5006540062365383,
              "E_x": -0.31242853593747866,
              "E_xc": 0.0
            },
            "energy_total": -0.5006540062365383,
            "energy_update": 0.00019709703155679215,
            "mo_residual": 0.012606514282877552,
            "wall_time": 2.584951922
          },
          {
            "energy_terms": {
              "E_ee": 0.3120042667385621,
              "E_eext": 0.00965658208414476,
              "E_el": -0.49069175814205057,
              "E_en": -1.0008467878514151,
              "E_kin": 0.5010302404189421,
              "E_next": -0.009999999999998685,
              "E_nn": 0.0,
              "E_nuc": -0.009999999999998685,
              "E_tot": -0.5006917581420492,
              "E_x": -0.31253605953228436,
              "E_xc": 0.0
            },
            "energy_total": -0.5006917581420492,
            "energy_update": 3.775190551091878e-05,
            "mo_residual": 0.005629130599314004,
            "wall_time": 2.449191938
          }
        ],
        "wall_time": 5.034251303
      },
      "success": true
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": true
  }
}
