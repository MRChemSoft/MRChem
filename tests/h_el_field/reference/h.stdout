                                                                      
                                                                      
**********************************************************************
***                                                                ***
***                                                                ***
***        __  __ ____   ____ _                                    ***
***       |  \/  |  _ \ / ___| |__   ___ _ __ ___                  ***
***       | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                 ***
***       | |  | |  _ <| |___| | | |  __/ | | | | |                ***
***       |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                ***
***                                                                ***
***       VERSION            0.3.0-alpha                           ***
***                                                                ***
***       Git branch         modernize_energy_calculation          ***
***       Git commit hash    297ecd98e048a26da678-dirty            ***
***       Git commit author  Stig Rune Jensen                      ***
***       Git commit date    Thu Mar 5 12:13:23 2020 +0100         ***
***                                                                ***
***       Magnar Bjorgve     <magnar.bjorgve@uit.no>               ***
***       Roberto Di Remigio <roberto.di.remigio@uit.no>           ***
***       Luca Frediani      <luca.frediani@uit.no>                ***
***       Gabriel Gerez      <gsa017@post.uit.no>                  ***
***       Stig Rune Jensen   <stig.r.jensen@uit.no>                ***
***       Peter Wind         <peter.wind@uit.no>                   ***
***                                                                ***
**********************************************************************

----------------------------------------------------------------------

 MPI processes         :      (no bank)                             1
 OpenMP threads        :                                            1
 Total cores           :                                            1
                                                                      
----------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

----------------------------------------------------------------------

 MRCPP version         : 1.2.0-alpha2
 Git branch            : modernize_energy_calculation
 Git commit hash       : ea07b65c8109b8937518-dirty
 Git commit author     : Magnar Bj√∏rgve
 Git commit date       : Wed Mar 4 22:18:16 2020 +0100

 Linear algebra        : EIGEN v3.3.7
 Parallelization       : OpenMP

----------------------------------------------------------------------


                                                                      
======================================================================
                       MultiResolution Analysis
----------------------------------------------------------------------
 polynomial order      : 5
 polynomial type       : Interpolating
----------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   16.00000    16.00000    16.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -16.00000   -16.00000   -16.00000 ]
 upper bounds          : [   16.00000    16.00000    16.00000 ]
 total length          : [   32.00000    32.00000    32.00000 ]
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***                     Initializing Molecule                      ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
======================================================================
                               Molecule
----------------------------------------------------------------------
 Charge                :                                            0
 Multiplicity          :                                            2
----------------------------------------------------------------------
    N    Atom          :              x              y              z
----------------------------------------------------------------------
    0       H          :       0.000000       0.000000       1.000000
----------------------------------------------------------------------
 Center of mass        :       0.000000       0.000000       1.000000
 Gauge origin          :       0.000000       0.000000       0.000000
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***              Computing Initial Guess Wavefunction              ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Diagonalize Hamiltonian matrix
 Precision             : 1.00000e-03
 Restricted            : False
 Hamiltonian           : Core
 AO basis              : Hydrogenic orbitals
 Zeta quality          : 2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
                          Molecular Orbitals
----------------------------------------------------------------------
 Alpha electrons       :                                            1
 Beta electrons        :                                            0
 Total electrons       :                                            1
----------------------------------------------------------------------
    n  Occ Spin        :                                         Norm
----------------------------------------------------------------------
    0    1    a        :                           1.000000000000e+00
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***              Computing Ground State Wavefunction               ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Compute initial energy
 Method                : Hartree-Fock
 Precision             : 1.00000e-03
 Localization          : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
                           Molecular Energy
----------------------------------------------------------------------
 Kinetic energy        :           (au)                0.500038592412
 E-N energy            :           (au)               -1.000003113631
 Coulomb energy        :           (au)                0.311494116621
 Exchange energy       :           (au)               -0.312479677884
 X-C energy            :           (au)                0.000000000000
 Ext. field (el)       :           (au)                0.010000015936
 Ext. field (nuc)      :           (au)               -0.010000000000
----------------------------------------------------------------------
 Electronic energy     :           (au)               -0.490950066545
 Nuclear energy        :           (au)               -0.010000000000
----------------------------------------------------------------------
 Total energy          :           (au)           -5.009500665453e-01
                       :     (kcal/mol)           -3.143509102534e+02
                       :       (kJ/mol)           -1.315244209354e+03
                       :           (eV)           -1.363154515159e+01
======================================================================


======================================================================           
                           Orbital Energies                                      
----------------------------------------------------------------------           
    n  Occ Spin        :                                      Epsilon            
----------------------------------------------------------------------           
    0    1    a        :           (au)               -0.491935627808            
----------------------------------------------------------------------           
 Sum occupied          :           (au)               -0.491935627808            
======================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Optimize ground state orbitals
 Method                : Hartree-Fock
 Max iterations        : 5
 KAIN solver           : 3
 Localization          : Off
 Diagonalization       : First two iterations
 Helmholtz precision   : 1.00000e-03
 Start precision       : 1.00000e-03
 Final precision       : 1.00000e-03
 Energy threshold      : 1.00000e-04
 Orbital threshold     : 1.00000e-02
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
 Iter         MO residual            Total energy              Update
----------------------------------------------------------------------
    0        1.000000e+00         -0.500950066545       -5.009501e-01
    1        1.284801e-02         -0.501161670398       -2.116039e-04
    2        5.759561e-03         -0.501202800110       -4.112971e-05
----------------------------------------------------------------------
                   SCF converged in 2 iterations!
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***                 Printing Molecular Properties                  ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
======================================================================
                               Molecule
----------------------------------------------------------------------
 Charge                :                                            0
 Multiplicity          :                                            2
----------------------------------------------------------------------
    N    Atom          :              x              y              z
----------------------------------------------------------------------
    0       H          :       0.000000       0.000000       1.000000
----------------------------------------------------------------------
 Center of mass        :       0.000000       0.000000       1.000000
 Gauge origin          :       0.000000       0.000000       0.000000
======================================================================


======================================================================
                           Orbital Energies
----------------------------------------------------------------------
    n  Occ Spin        :                                      Epsilon
----------------------------------------------------------------------
    0    1    a        :           (au)               -0.492250268397
----------------------------------------------------------------------
 Sum occupied          :           (au)               -0.492250268397
======================================================================


======================================================================
                           Molecular Energy
----------------------------------------------------------------------
 Kinetic energy        :           (au)                0.502858217557
 E-N energy            :           (au)               -1.002664607548
 Coulomb energy        :           (au)                0.312455661259
 Exchange energy       :           (au)               -0.313503129546
 X-C energy            :           (au)                0.000000000000
 Ext. field (el)       :           (au)                0.009651058167
 Ext. field (nuc)      :           (au)               -0.010000000000
----------------------------------------------------------------------
 Electronic energy     :           (au)               -0.491202800110
 Nuclear energy        :           (au)               -0.010000000000
----------------------------------------------------------------------
 Total energy          :           (au)           -5.012028001105e-01
                       :     (kcal/mol)           -3.145095029586e+02
                       :       (kJ/mol)           -1.315907761233e+03
                       :           (eV)           -1.363842238194e+01
======================================================================


======================================================================
                            Dipole Moment
----------------------------------------------------------------------
 Electronic vector     :      -0.000000      -0.000000      -0.965106
 Magnitude             :           (au)                      0.965106
                       :        (Debye)                      2.453054
----------------------------------------------------------------------
 Nuclear vector        :       0.000000       0.000000       1.000000
 Magnitude             :           (au)                      1.000000
                       :        (Debye)                      2.541746
----------------------------------------------------------------------
 Total vector          :       0.000000       0.000000       0.034894
 Magnitude             :           (au)                      0.034894
                       :        (Debye)                      0.088692
======================================================================


                                                                      
                                                                      
**********************************************************************
***                                                                ***
***                         Exiting MRChem                         ***
***                                                                ***
***                    Wall time :  0h  0m  7s                     ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
